### ISSUE TITLE
at-rule-no-unknown false positive for less

### ISSUE THREAD

**Issue by ssivanatarajan** [stylelint/stylelint#5210]
> Clearly describe the bug

There are false positive for at-rule-unknown rule in less , if there is no space between proprty and variable

> Which rule, if any, is the bug related to?

`at-rule-unknown `

> What code is needed to reproduce the bug?

```less
.a() {
    @box-shadow:none !important;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
     "at-rule-no-unknown": true,
  }
}
```

> Which version of stylelint are you using?

e.g. `13.12.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

 "Node JS API"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

 "Yes, it's related to Less  properties"

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

 "The following warnings were flagged:"

```Unexpected unknown at-rule "@box-shadow:none"
```

**Comment by jeddy3** [comment]
@ssivanatarajan Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACA3AhgJ3ugNgVxgAIBCASwFsAHCDAFzQDtaBuAHQZABoQBnATyZoAHghA4YPHl3AQGAM1IBzUcHaFCrEBjziem+IVUN16zWloBabeIsMIFvAwDWdgO4N9hWtpic1JzQAjHAgwJ1t7GCpaPk9vAj9jAJkQjAiLUgZcUigLAAsYITifRJMNGXJyGCZ0qMoY4oT-UxBYMBxMc1JZC2DQ8LsLKDxKHFIwcxgLMDweWghyC0oMCEoYOlIJRt9m8raOjC6evrD04dHxyaWVtY2thEIAbV31eJ2ksqMyss0lOwwYJ5nh9vi1IAweDAZrRSCgpucxhNaBILK5SLQ8kNSHI5OtqpZcAQ9CAXiYALqkgC+uzJpWS+06MOOIVOgx4eRoGMYuWWq3WMQsEDhGAwOXuBjedJaclkljkaHIpBwfDOI0RVwYCvFXhKu00Mpq8sVyvSiskmUUFkU1XW4wsRqVKqcMD4rhoUG2UvKckcYCZDGmaBwYHSmWyHoekr1IB9DD93QDE2D6UcPEoaDAMFyfMO8wwnujsfjPTGDBgmCthygmxqgzsEPoDCgmFy1YBxY8kd1IM0zr4ckOVSGUIORwDgwo1DojFoBZ7ICq1bQ9vLtDwANsWpTzjcnYl3Z+IE1i8rOQsmHLPFDWSDOTnh8GsB4mabFosaah2PG6NiXaa88fNVLmRc9LEnTkrB0bUowA+wEWAqZIXEP0aGJfd-wfSJohVHgIHXTN72SQZCm8ZdIUVSAQghQiWgnG8xlbPDgkQjp2WmBYqiYNCdQw5JeVuAVBkcFwIHcGjyiQqE8yWSE8Cgex2jQSRtxEsS-3eQ9JJQtJKFk+SLBgcROMsISd1EvceI05ItOkmI1hU3dxM0OZRQYS06xgVxS0BdSvU0Rx0Qc8ynJAcxIJsOoBW83oYBlAEgVJMwcFcNA+GJL11C+UFykKTN6gSkFQSCFknF0K80DkZE0h4LdjymE5SokdLSUPBQMDmWwJGRCMWsIWlet+RR-h8gxgWy5IKqq9jKnxTReopQrCGpEF+vnMLrCmOrA0hTxNBCVx1k0PzQssDbNyHNMMymSbDoeJKUrS98LRsbyjujdaoPfKJxggKjOq8zJrsq26DHu1LmvnBrphCHg30CQ5MwM7CLGiwJYpoEbyjLYU3shkroYgWG3N6BHNs86KbvzO7QuS8HccPKH2kJuHSf+1H0fi6mgweq9yB0GEUcB+nkkZmGWau9900RtG4sxsHHqJxQXqFrhoyhvkGHFxGywBstz2BqnQZpnmLD5nABde1W8f6QU1k14n4Yly7Ecp3bjfBp63OVsthZadW7a1xCpfqjm5fdtLffKSiaHyQptrD-bbuO6O0gKIQUeqRQMTd9lOUjzRIBmmpIpVdnZYKorw4h0Essr3KYHyh4xvG71SHakyuqzObFvUVbxsG4aK5b5yYnEUtLEL8huWJeaqRpZOOPxVE8nRCRg4yCExTd7m6atw8ZjmBZrhzAUS8FvWZYxofDx3iO9++Wvvk0evG9G3qWkp6ZZnmRZ+P5X8MpPxjG3DqZY5hdxJD3PqgCWh-Cvk3d+5RP6T1mvffuIBMiw1gJ7JWUxUYlW7uNBaoJlplD7vSEcjIEy9EYJaZ2QMqpu2xknaMDJczUMCLQyWEtL6cyNrfauFDFLsOZDbciP0-o6wpgbbetNHpmwtirY6bCxy9HxuI6OAZ6H60YVzORV5Fbex8soyhIiAxQw0b9Ho2jeFh2YVTExwjVEWOengwGpthDDicf6biABGRxo5-RqJtqRJUb5LFUVkTzfOrRTGqM0Wzdxrs9EmwUaQc+xjWFxKCQk7RyT+H6JwUYmJKiclWK0evWxTCYA4zQXsbJ1Cz5l3gW-Ra8tBGfAGiAF+s4EFQI-gbLxgSEyR0rm1UBncepQPIZXOBfCniILMIMlBTBRlAMwWKIpbiL4EMgUQueK1jpFjKZUZcUikkyJSR7NJGSYnHOoZPMi698lIMKYY7ZmT5z3J6I87h0tQ7VNqUc30QSp7pyaYDbiAAGYFcYglbQmDtamicHGFhBdQ9MAImAFEhFec5esNmwCidc-m6TLawo7EsC82KJAom0YSuxNT1hbNuXU-U6KeiuBXsiPJlyCnRLZRgps+IxyeAAEzHXEG5DEFgABe6x7CmXRCFMFSN6il0hZ4fx0ZFykGXLicw64pi5OeXy15Arjq6v1auI17E-o2IBdTexMSrUrkNRuBFSkE6iRYfOV1Bq1wbkxfiHFdL14MsBb6w8-qbUbkOG5KYOZzAx15bo-lu9LVZj1W6wNUx43WltusZN1VKmOvTXfTNS4LAAEcCAYA1R1X5+KGEg3NSS82ZKlE6qzcuWt6wG0Tw4k8iWLz2ksvJd2qtfb62C0bUOv5Idy5OqZai2CBlfrL1XvQkKgwzSKwMk2QUch3x4QwNrcmKt0JWVongcgaM0jiDQNWYm8qVjEordGBgt773pFCaWS0r7CYhT-nQFUiLvUHVXYeM6EKL5loWW0quptSWssAY-Q8PTr7ZX1CAju4CplEJgeUOZmNm7oOQYvVZezsrEO+KQ8kx0bIx3MN4UggQ8AgUdmEGAtADHhs3kS5dQLoxMbSCx0U7GQJJukqm1tmhnWCtE6BVjknE23GLQumKS6jYKcY4ZKSMdC6BEyBp2Ths20fvnEpozJmZOlu0xZjpEl9PaWHM+aozYag2c1NJOs1jg4hSU2MOdpzEl61HVXGJQXSAhanppqpQmo3WRc9JVVsHtQwpEylmOukYByQUqxK84G3Yoqi9lnSekCtKSvMGmluLNMRsS1B5LyFpK5fywZIyS9ivIp9c1loSn2v6X08ZO1PQ7D0H9G7eSeBmJldazHOyxqvUlb6zEgKE8Vu9cgzEwkUw+xugwLkHrRtSuCr27OwdoXm06Lk0hm5E75wXeC1duLZn30GNcah6Mz2YuveHf8hz8mV27aDAQS7HjwXI28tC-wy1KQgEpEAA) with:

```less
@var:value !important;
```

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes#working-on-syntaxes) if you have time.

**Comment by ssivanatarajan** [comment]
@jeddy3  so do i need to fix this issue in https://github.com/shellscape/postcss-less ?

**Comment by jeddy3** [comment]
I believe so, although not 100% sure. You'll need to test how the parser parses:

```
@var:value !important;
```

If the AST doesn't look correct, you should report an issue with that package. The author of the parser usually welcomes pull requests to fix issues like this.
